ecosystem/python/sdk/aptos_sdk/async_client.py

```diff
@@ -780,19 +780,22 @@ class FaucetClient:
     base_url: str
     rest_client: RestClient
 
-    def __init__(self, base_url: str, rest_client: RestClient):
+    def __init__(self, base_url: str, rest_client: RestClient, auth_token: str = None):
         self.base_url = base_url
         self.rest_client = rest_client
+        self.auth_token = auth_token
 
     async def close(self):
         await self.rest_client.close()
 
     async def fund_account(self, address: AccountAddress, amount: int):
         """This creates an account if it does not exist and mints the specified amount of
         coins into that account."""
-        response = await self.rest_client.client.post(
-            f"{self.base_url}/mint?amount={amount}&address={address}"
-        )
+        request = f"{self.base_url}/mint?amount={amount}&address={address}"
+        headers = {}
+        if self.auth_token:
+            headers["Authorization"] = f"Bearer {self.auth_token}"
```

Comment:
a more suitable approach might be to place the headers into the object during construction (in `__init__`) so we don't reconstruct them here each time.